airbyte-integrations/connectors/source-google-search-console/source_google_search_console/streams.py

```diff
@@ -0,0 +1,289 @@
+    ) -> Optional[Union[Dict[str, Any], str]]:
+
+        start_date = self._get_start_data(stream_state, stream_slice)
+
+        data = {
+            "startDate": start_date,
+            "endDate": self._end_date,
+            "dimensions": self.dimensions,
+            "searchType": stream_slice.get("search_type"),
+            "aggregationType": "auto",
+            "startRow": self.start_row,
+            "rowLimit": ROW_LIMIT,
+        }
+        return data
+
+    def _get_start_data(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+    ) -> str:
+        start_date = self._start_date
+
+        if start_date and stream_state:
+            if stream_state.get(unquote_plus(stream_slice["site_url"]), {}).get(stream_slice["search_type"]):
+                stream_state_value = stream_state.get(unquote_plus(stream_slice["site_url"]), {}).get(stream_slice["search_type"])
+
+                start_date = max(
+                    pendulum.parse(stream_state_value[self.cursor_field]),
+                    pendulum.parse(start_date),
+                ).to_date_string()
+
+        return start_date
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        records = response.json().get(self.data_field) or []
+
+        for record in records:
+            record["site_url"] = unquote_plus(stream_slice.get("site_url"))
+            record["search_type"] = stream_slice.get("search_type")
+
+            for dimension in self.dimensions:
+                record[dimension] = record["keys"].pop(0)
+
+            # remove unnecessary empty field
+            record.pop("keys")
+
+            yield record
+
+    def get_updated_state(
+        self,
+        current_stream_state: MutableMapping[str, Any],
+        latest_record: Mapping[str, Any],
+    ) -> Mapping[str, Any]:
+        latest_benchmark = latest_record[self.cursor_field]
+
+        site_url = latest_record.get("site_url")
+        search_type = latest_record.get("search_type")
+
+        if current_stream_state.get(site_url, {}).get(search_type):
+            current_stream_state[site_url][search_type] = {
+                self.cursor_field: max(latest_benchmark, current_stream_state[site_url][search_type][self.cursor_field])
+            }
+
+        elif current_stream_state.get(site_url):
+            current_stream_state[site_url][search_type] = {self.cursor_field: latest_benchmark}
+
+        else:
+            current_stream_state = {site_url: {search_type: {self.cursor_field: latest_benchmark}}}
+
+        # this line is required to pass the acceptance test successfully
+        current_stream_state[self.cursor_field] = current_stream_state[site_url][search_type][self.cursor_field]
```

Comment:
is there any way to avoid having this here i.e; adding information only to use it in tests? Is the problem that we want the max over all search types and the yaml format doesn't support that? if so let's add this comment to help the next dev understand 
```suggestion
        # we need to get the max date over all searchTypes but the current acctest YAML format doesn't 
        # support that
        current_stream_state[self.cursor_field] = current_stream_state[site_url][search_type][self.cursor_field]
```